api/src/main/java/org/openmrs/api/context/Context.java

```diff
@@ -576,6 +578,8 @@ public static ProgramWorkflowService getProgramWorkflowService() {
 	 */
 	public static MessageService getMessageService() {
 		MessageService ms = getServiceContext().getMessageService();
+		AbstractApplicationContext context = new ClassPathXmlApplicationContext("applicationContext-service.xml");
+		ms = (MessageService) context.getBean("messageServiceBean", MessageService.class);
```

Comment:
Can you walk me through what these two lines do and how they differ from the existing line 580?